.github/workflows/pulumi-cli.yml

```diff
@@ -0,0 +1,102 @@
+env:
+  GITHUB_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
+
+name: cli docs build
+on:
+  pull_request:
+  repository_dispatch:
+    types:
+      - pulumi-cli
+jobs:
+  pull-request:
+    runs-on: ubuntu-latest
+    needs: build-pulumi-cli-docs
+    steps:
+      - name: checkout docs repo
+        uses: actions/checkout@v2
+      - name: pull-request
+        uses: repo-sync/pull-request@v2
+        with:
+          source_branch: "pulumi/${{ github.run_id }}-${{ github.run_number }}"
+          destination_branch: "master"
+          pr_title: "Regen docs pulumi@${{ github.event.client_payload.ref }}"
+          pr_body: ""
+          pr_label: "automation/pulumi-cli-docs"
+          github_token: ${{ secrets.PULUMI_BOT_TOKEN }}
+
+  build-pulumi-cli-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: set the pulumi version
+        run: |
+          echo "PULUMI_VERSION=2.12.0" >> $GITHUB_ENV
+          #echo "{PULUMI_VERSION}={${{ github.event.client_payload.ref }}}" >> $GITHUB_ENV
+      - name: checkout docs repo
+        uses: actions/checkout@v2
+        with:
+          path: docs
+      - name: checkout pulumi repo
+        uses: actions/checkout@v2
+        with:
+          repository: pulumi/pulumi
+          path: pulumi
+          ref: ${{ github.event.client_payload.ref }}
+      - name: Install pulumictl
+        uses: jaxxstorm/action-install-gh-release@d62853bc9e466834c74cefb34fad78535582e21a
+        with:
+          repo: pulumi/pulumictl
+      - name: Install Pulumi CLI
+        uses: pulumi/action-install-pulumi-cli@b811287a6e06bb7e5d8b91132b839c67e3f9c6ed
+        with:
+          pulumi-version: ${{ env.PULUMI_VERSION }}
```

Comment:
It can't be env.PULUMI_VERSION unless you set that from the event details right?

Reply:
it'll be extracted from the payload ref here, we need to strip the `refs` from the payload.